pkg/checks/traceroute/traceroute.go

```diff
@@ -109,13 +148,24 @@ func readIcmpMessage(ctx context.Context, icmpListener *icmp.PacketConn, timeout
 
 // TraceRoute performs a traceroute to the specified host using TCP and listens for ICMP Time Exceeded messages using ICMP.
 func TraceRoute(ctx context.Context, cfg tracerouteConfig) (map[int][]Hop, error) {
-	// maps ttl -> attempted hops for that ttl
+	span := trace.SpanFromContext(ctx)
+	tr := span.TracerProvider().Tracer("tracer.traceroute.trace")
+	ctx, sp := tr.Start(ctx, "TraceRoute", trace.WithAttributes(
+		attribute.String("target", cfg.Dest),
+		attribute.Int("port", cfg.Port),
+		attribute.Int("max_hops", cfg.MaxHops),
+		attribute.Stringer("timeout", cfg.Timeout),
+	))
+	defer sp.End()
+
+	// Maps ttl -> attempted hops for that ttl
 	hops := make(map[int][]Hop)
 	log := logger.FromContext(ctx).With("target", cfg.Dest)
 
 	addr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", cfg.Dest, cfg.Port))
 	if err != nil {
-		log.Error("failed to resolve target name", "err", err.Error())
+		sp.RecordError(err)
```

Comment:
`SetStatus` is missing here, so we know that this span (and not only the child spans) has an error.